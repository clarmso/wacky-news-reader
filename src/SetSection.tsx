import { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import {
  Grid,
  Select,
  FormControl,
  InputLabel,
  MenuItem,
  Link,
} from "@material-ui/core";
import { WackyState } from "./connect/reducer";
import { sections } from "./Page";
import { fetchArticles } from "./connect/actions";

const SetSection: React.FC = () => {
  const dispatch = useDispatch();
  const currentSection = useSelector((state: WackyState) => state.section);

  const [section, setSection] = useState(currentSection);
  const [openSelect, setOpenSelect] = useState(false);

  const handleSelectChange = (event: React.ChangeEvent<{ value: unknown }>) => {
    const newSection = event.target.value as string;
    setSection(newSection);
    dispatch(fetchArticles(newSection));
  };
  const handleSelectOpen = () => {
    setOpenSelect(true);
  };
  const handleSelectClose = () => {
    setOpenSelect(false);
  };

  return (
    <Grid container spacing={5} justify="center">
      <Grid item>
        <FormControl>
          <InputLabel>Section</InputLabel>
          <Select
            open={openSelect}
            onClose={handleSelectClose}
            onOpen={handleSelectOpen}
            value={section}
            onChange={handleSelectChange}
            data-cy="section-menu"
          >
            {Object.values(sections).map((section: string) => {
              return (
                <MenuItem
                  value={section}
                  key={`menu-item-${section}`}
                  data-cy={`menu-item-${section}`}
                >
                  {section}
                </MenuItem>
              );
            })}
          </Select>
        </FormControl>
      </Grid>
    </Grid>
  );
};

export default SetSection;
