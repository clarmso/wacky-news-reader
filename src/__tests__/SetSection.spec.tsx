import { Provider } from "react-redux";
import { createStore } from "redux";
import { render } from "@testing-library/react";
import SetSection from "../SetSection";
import wackyReducer, { initialState } from "../connect/reducer";

it("renders with initial state", () => {
  const store = createStore(wackyReducer, initialState);
  const { asFragment } = render(
    <Provider store={store}>
      <SetSection />
    </Provider>
  );
  const renderedHtml = asFragment();
  expect(renderedHtml).toMatchSnapshot();
});
