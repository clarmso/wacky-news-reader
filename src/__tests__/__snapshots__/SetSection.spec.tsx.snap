// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with initial state 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          Section
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-menuItem-1 MuiInputBase-formControl MuiInput-formControl"
          data-cy="section-menu"
        >
          <div
            aria-haspopup="listbox"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            role="button"
            tabindex="0"
          >
            world
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            tabindex="-1"
            value="world"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
